test/jest/victory-bar/victory-bar.spec.js

```diff
@@ -0,0 +1,257 @@
+      const svg = container.querySelector("svg");
+      expect(svg.getAttribute("style")).toContain("width: 100%; height: 100%");
+    });
+
+    it("renders an svg with the correct viewBox", () => {
+      const { container } = render(<VictoryBar />);
+      const svg = container.querySelector("svg");
+      const viewBoxValue = `0 0 ${450} ${300}`;
+      expect(svg.getAttribute("viewBox")).toEqual(viewBoxValue);
+    });
+
+    it("renders 4 bars", () => {
+      const { container } = render(<VictoryBar />);
+      const bars = container.querySelectorAll("path");
+      expect(bars.length).toEqual(4);
+    });
+
+    it("renders each bar as a rectangle", () => {
+      const { container } = render(<VictoryBar />);
+      const barCommandStrings = Array.from(
+        container.querySelectorAll("path")
+      ).map((bar) => bar.getAttribute("d"));
+      barCommandStrings.forEach(svgExpectations.expectIsRectangular);
+    });
+  });
+
+  describe("rendering data", () => {
+    it("renders bars for {x, y} shaped data (default)", () => {
+      const data = range(10).map((i) => ({ x: i, y: i }));
+      const { container } = render(<VictoryBar data={data} />);
+      const bars = container.querySelectorAll("path");
+      expect(bars.length).toEqual(10);
+    });
+
+    it("renders ordered bars when sortKey is passed", () => {
+      const data = range(5)
+        .map((i) => ({ x: i, y: i }))
+        .reverse();
+      const { container } = render(<VictoryBar data={data} sortKey="x" />);
+      const barHeight = Array.from(container.querySelectorAll("path")).map(
+        (bar) => {
+          const commandString = bar.getAttribute("d");
+          return getShapeUtils.getBarHeight(commandString);
+        }
+      );
+
+      const ascendingBars = [...barHeight].sort((a, b) => a - b);
+
+      expect(barHeight).toEqual(ascendingBars);
+    });
+
+    it("renders reverse ordered bars when sortOrder is descending", () => {
+      const data = range(5)
+        .map((i) => ({ x: i, y: i }))
+        .reverse();
+      const { container } = render(
+        <VictoryBar data={data} sortKey="x" sortOrder="descending" />
+      );
+      const barHeight = Array.from(container.querySelectorAll("path")).map(
+        (bar) => {
+          const commandString = bar.getAttribute("d");
+          return getShapeUtils.getBarHeight(commandString);
+        }
+      );
+      const descendingBars = [...barHeight].sort((a, b) => b - a);
+
+      expect(barHeight).toEqual(descendingBars);
+    });
+
+    it("renders bars for array-shaped data", () => {
+      const data = range(20).map((i) => [i, i]);
+      const { container } = render(<VictoryBar data={data} x={0} y={1} />);
+      const bars = container.querySelectorAll("path");
+      expect(bars).toHaveLength(20);
+    });
+
+    it("renders bars for deeply-nested data", () => {
+      const data = range(40).map((i) => ({ a: { b: [{ x: i, y: i }] } }));
```

Comment:
Tiny nitpick - we can probably render fewer bars here. I don't know whether that will have any impact on speed, but we're not getting any benefit to rendering 40 bars here, the test would work just as well if there were 4.